core/src/main/scala/org/apache/spark/HttpServer.scala

```diff
@@ -155,6 +156,7 @@ private[spark] class HttpServer(
       throw new ServerStateException("Server is already stopped")
     } else {
       server.stop()
+      Option(server.getThreadPool).collect { case x: LifeCycle => x }.foreach(_.stop())
```

Comment:
As I said, I don't think we are going for idiomatic Scala in Spark. Actually in most cases we go for readability.